api/api/v1alpha1/manifest_types.go

```diff
@@ -35,21 +35,37 @@ type CustomState struct {
 	State      string `json:"state,omitempty"`
 }
 
-// ChartInfo defines chart information
-type ChartInfo struct {
-	ChartPath    string `json:"chartPath,omitempty"`
-	RepoName     string `json:"repoName,omitempty"`
-	Url          string `json:"url,omitempty"`
-	ChartName    string `json:"chartName,omitempty"`
-	ReleaseName  string `json:"releaseName,omitempty"`
-	ClientConfig string `json:"clientConfig,omitempty"`
-	Overrides    string `json:"overrides,omitempty"`
+// InstallInfo defines installation information
+type InstallInfo struct {
+	ConfigInfo `json:",inline"`
+	// +kubebuilder:validation:Enum=HelmChart;Executable
+	Type string `json:"type"`
+	Name string `json:"name"`
+}
+
+// ConfigInfo defines config information
+type ConfigInfo struct {
+	Repo   string `json:"repo"`
+	Module string `json:"module"`
+	Digest string `json:"digest"`
```

Comment:
I just double checked the fetching process, and we will also need to include a Version to reference the Digest correctly since the Digest Scope is not Artifact but Version-Scoped.